src/TimeSeries.svelte

```diff
@@ -0,0 +1,692 @@
+    const month = new Intl.DateTimeFormat("en-US", { month: "2-digit" }).format(
+      date
+    );
+    const day = new Intl.DateTimeFormat("en-US", { day: "2-digit" }).format(
+      date
+    );
+    const year = new Intl.DateTimeFormat("en-US", { year: "numeric" }).format(
+      date
+    );
+
+    return `${year}-${month}-${day}`;
+  }
+
+  function convertStorageDate(storageDate: string): Date {
+    const date = new Date(storageDate);
+    const time = date.getTime();
+    const timeZoneOffset = date.getTimezoneOffset() * 60 * 1000;
+    const adjustedTime = time + timeZoneOffset;
+
+    return new Date(adjustedTime);
+  }
+
+  type HtmlSelection = d3.Selection<HTMLElement, any, any, any>;
+  type SvgSelection = d3.Selection<SVGGElement, any, any, any>;
+  type ElementSection = HtmlSelection | SvgSelection;
+
+  function generateChart() {
+    const chartAreaContainerElement: SVGElement = chartContainerElement.querySelector(".chart-area-container");
+    const chartAreaElement: SVGGElement = chartAreaContainerElement.querySelector("g");
+    const xElement: SVGGElement = chartContainerElement.querySelector(".x.axis");
+    const yElement: SVGGElement = chartContainerElement.querySelector(".y.axis");
+    const pathsElement: SVGGElement = chartContainerElement.querySelector(".paths");
+    const verticalLineElement: SVGGElement = chartContainerElement.querySelector(".chart-vertical-line");
+    let chartArea: SvgSelection;
+    let x: SvgSelection;
+    let y: SvgSelection;
+    let paths: SvgSelection;
+    let verticalLine: SvgSelection;
+
+    let chartWidth = chartAreaContainerElement.getBoundingClientRect().width;
+
+    if (chartAreaElement) {
+      chartArea = d3.select(chartAreaElement);
+    } else {
+      chartArea = d3
+        .select(chartAreaContainerElement)
+        .attr(
+          "viewBox",
+          [
+            -chartBounds.margin,
+            0,
+            chartBounds.width + chartBounds.margin,
+            chartBounds.height + chartBounds.margin,
+          ].join(" ")
+        )
+        .append("g");
+    }
+
+    if (xElement) {
+      x = d3.select(xElement);
+    } else {
+      x = chartArea
+        .append("g")
+        .attr("class", "x axis")
+        .attr("transform", `translate(0, ${chartBounds.height})`);
+    }
+
+    if (yElement) {
+      y = d3.select(yElement);
+    } else {
+      y = chartArea.append("g").attr("class", "y axis");
+    }
+
+    if (pathsElement) {
+      paths = d3.select(pathsElement);
+    } else {
+      paths = chartArea.append("g").attr("class", "paths");
+    }
+
+    if (verticalLineElement) {
+      verticalLine = d3.select(verticalLineElement);
+    } else {
+      verticalLine = chartArea
+        .append("g")
+        .append("line")
+        .attr("class", "chart-vertical-line inactive")
+        .attr("y1", 0)
+        .attr("y2", chartBounds.height);
+    }
+
+    let data: RegionalTrends[] = Array.from(regionalTrendsByPlaceId.values()).filter(
```

Comment:
I'm getting an error in the console, but it seems to be a minor issue because the chart still works OK.  Can you check to see if there's an initialization guard that you can add to prevent this error?

```
TimeSeries.svelte:486 Uncaught (in promise) TypeError: Cannot read property 'values' of undefined
    at generateChart (TimeSeries.svelte:486)
    at TimeSeries.svelte:647
    at Object.subscribe (index.mjs:51)
    at TimeSeries.svelte:642
    at Generator.next (<anonymous>)
    at TimeSeries.svelte:415
    at new Promise (<anonymous>)
    at __awaiter (TimeSeries.svelte:415)
    at TimeSeries.svelte:640
    at run (index.mjs:18)
```